### ISSUE TITLE
Completion provider ranking/grouping doesn't seem to work

### ISSUE THREAD

**Issue by segevfiner** [Microsoft/vscode#63877]
- VSCode Version: 1.29.1 & Git build at 347216b6120cd31b7ba9393e47b8de8fb585b6b3
- OS Version: Ubuntu 18.04.1 amd64 / macOS 10.14.1

Steps to Reproduce:

1. Save the following as `.circleci/config.yml` in a new folder and open the folder in VS Code:
```yaml
version: 2

jobs:
  build:
    docker:
      - image: python:3.7

    steps:
      - 
```
2. Trigger completions after the dash that is after `steps`.
3. You are likely to get word completions instead of the completions from `redhat.vscode-yaml`.

![screenshot_2018-11-27_20-48-08](https://user-images.githubusercontent.com/24731903/49104599-0567de00-f287-11e8-8ef1-414450099e86.png)

### Analysis
I think the issue is around here [src/vs/editor/contrib/suggest/suggest.ts:115](https://github.com/Microsoft/vscode/blob/2f9f0c35ccb681f9a1ff53e7619677b00a23cb95/src/vs/editor/contrib/suggest/suggest.ts#L115). I think this code stops on *any* provider that is not the snippet provider, no matter it's rank/group, while it should be letting the higher ranked providers finish generating completions, and only show suggestions from the providers in the highest provider group that supplies them (Except the snippets group which should always show mixed with the highest ranked group's completions).

Does this issue occur when all extensions are disabled?: Yes (Except redhat.vscode-yaml)

Originally reported at: https://github.com/redhat-developer/vscode-yaml/issues/139

cc @JPinkney @fbricon

**Comment by jrieken** [comment]
> I think this code stops on any provider that is not the snippet provider, no matter it's rank/group,

Nope, the order is snippets, group of best ranked providers, group of 2nd best ranked providers etc, etc. Word based completion is always is the lowest group and only asked when no other provider has provided a result (and snippets don't count for that). 

This surely is an issue with the vscode-yaml extension, it's likely throwing an error or not returning a result.

**Comment by segevfiner** [comment]
Some extra info that I posted on the vscode-go like issue.

Using:
VS Code 1.29.1
Xubuntu 18.04.1 amd64 and also on macOS 10.14.1
vscode-yaml 0.1.0

Log: [2-Yaml Support.log](https://github.com/redhat-developer/vscode-yaml/files/2621569/2-Yaml.Support.log)

In case it's some other extension breaking this, here are the extensions I have on the Linux box:
```
bungcip.better-toml@0.3.2
dbaeumer.vscode-eslint@1.7.0
eamodio.gitlens@8.5.6
lextudio.restructuredtext@83.0.0
ms-python.python@2018.10.1
ms-vscode.cpptools@0.20.1
ms-vscode.Go@0.7.0
ms-vscode.vscode-typescript-tslint-plugin@0.1.0
msjsdiag.debugger-for-chrome@4.11.1
redhat.vscode-yaml@0.1.0
stkb.rewrap@1.9.1
```

The attached log shows that the language server is generating completions correctly. There is  also the option of it getting lost along the way or maybe being malformed somehow... I will try testing this further when I have the time.
